File: Programmers/자바기초/2의영역.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        
        ArrayList<Integer> idxList = new ArrayList<>();
        for (int i = 0 ; i < arr.length ; i++) {
            if (arr[i] == 2) {
                idxList.add(i);
            }
        }
        
        if (idxList.size() == 0) {
            int[] answer = {-1};
            return answer;
        }
        
        int startIdx = idxList.get(0);
        int endIdx = idxList.get(idxList.size()-1);
        
        int[] answer = new int[endIdx-startIdx+1];
        
        int idx = 0;
        for (int i = startIdx ; i <= endIdx ; i++) {
            answer[idx] = arr[i];
            idx++;
        }
        
        return answer;
    }
}

File: Programmers/제로베이스_문제추천_py/레벨0/2의영역.py
def solution(arr):
    if 2 not in arr :
        answer = [-1]
    else :
        answer = []
        i_arr = []
        for i in range(len(arr)) :
            if arr[i] == 2 :
                i_arr.append(i)
        answer = arr[i_arr[0]:i_arr[-1]+1]
    return answer